x-pack/platform/plugins/shared/alerting/server/task_runner/task_runner.ts

```diff
@@ -520,6 +526,15 @@ export class TaskRunner<
         getDecryptedRule(this.context, ruleId, spaceId)
       );
 
+      // Check that this task is current
+      const scheduledTaskId = ruleData.rawRule.scheduledTaskId;
+      if (this.taskInstance.id !== ruleId && this.taskInstance.id !== scheduledTaskId) {
```

Comment:
I'm not sure we need to test for `this.taskInstance.id !== ruleId`.  By default, new rules will end up having the task id == rule id, but it's not required.  I think all we need is ` this.taskInstance.id !== scheduledTaskId`.  For my zombie task hunter, I check both the rule's task id is consistent and the task's rule id is consistent (in it's params), but we don't need to make that check (task's rule id), since we loaded the rule from the task.

Reply:
Okay I will remove it, thanks! I think I had that ruleId check first and added `this.taskInstance.id !== scheduledTaskId` to cover an additional case I came across in testing, so maybe it catches everything.

Reply:
I tried removing the `this.taskInstance.id !== ruleId` sub-expression, and my simple repro test (export an enabled rule, then re-import it with defaults) - in this case it appears to delete the task, which is not what we want to do (we want to disable in this case, instead).  

So I think leaving the expression as it is now is correct.